scenes/gameplay/map/map.gd

```diff
@@ -1,4 +1,15 @@
 extends Node2D
 
+onready var map_container = $Map
+var scale_map := Vector2.ZERO
 
-
+func _ready() -> void:
+	get_tree().connect("screen_resized", self, "_on_screen_resized")
+	
+func _on_screen_resized():
+	scale_map.x = Game.size.x / map_container.texture.get_size().x
+	scale_map.y = Game.size.y / map_container.texture.get_size().y
+	map_container.set_position(Vector2(Game.size.x/2, Game.size.y/2))
+	map_container.set_scale(Vector2(scale_map.x, scale_map.y))
+	
```

Comment:
Actually there is a global project setting to enable this behaviour in Godot games. 

There are a lot of insights about this on the official doc: https://docs.godotengine.org/en/stable/tutorials/viewports/multiple_resolutions.html